### DISCUSSION TITLE
View の構造を把握しやすくするために目立つコメントを記述する。

### DISCUSSION THREAD

**Discussion by YusukeHosonuma** [YusukeHosonuma/Effective-SwiftUI#25] [Ideas]
## Overview

プログラミング言語のソースにおいてコメントは、処理の内容を説明・補足したり（What?）、それをなぜ行っているか（Why?）、を記述するのに利用されることが多い。

しかし、コメントは通常エディタによってシンタックスハイライトされ、**他のソースとは異なるもの**として認識され、構造を把握しやすくするための**アイキャッチ**としても機能する。

以下は、目立つような絵文字コメントを View に記述した例である。

![image](https://user-images.githubusercontent.com/2990285/162369256-3a6f7416-b29e-4f4d-99ae-ef4a4d42b4b1.png)

あなたはこのコードを初めて読むはずだが、全体の構造が素早く把握できたのではないだろうか。

## Discussion

このあたりは好き嫌いが分かれるため、チームで開発している場合は、どこまで利用するか事前に話し合うこと。

絵文字は強力なアイキャッチとして機能する一方で、どのような絵文字を利用するかを考える手間などもかかる。また、不統一な絵文字の乱用は、ときに誤解を招くため、十分に注意してプロジェクトに適用すること。

**Comment by uhooi** [comment]
アイキャッチに絵文字を使うの、私はやったことがなかったのですが、わかりやすくてよさそうですー！

ただ個人的にですが、コメントの上下にも空白のコメントを入れると、一画面に表示されるコードが減って構造を把握しづらくなるので、入れないほうがいいと思いました！
（好みの問題にはなると思いますが…）

**Comment by yuukiw00w** [comment]
コードでわかることはコードに任せる方がいいかなと思いました
例示されたコードの場合、accountのセクションであることはコードから読み取れるので、コメントで書いてしまうと二重管理になり、たとえば表示されるテキストが変わる場合にコメントも変える必要が出てきます

全体の構造がわかるようにしたい場合、全体構造がわかる程度の大きさになるよう詳細部分は別で切り出してあげるとコメントがなくても構造がわかりやすくなると思います

```swift
let accountSection = Section("account") { ... }
let linksSection = Section("links") { ... }

// ↑で詳細を定義して、↓に大きな構造だけ書く
List {
    accountSection
    linksSection
}.sheet(isPresented: $isPresentedLoginView) {
    LoginView()
}.confirmationDialog( ... ) {
    ...
}
```
